### ISSUE TITLE
[2022.05] PHP Warnings

### ISSUE THREAD

**Issue by tobiasd** [friendica/friendica#11217]
Please report here any PHP Warning message obtained running the 2022.02 stable or 2022.05-dev versions.

Replaces https://github.com/friendica/friendica/issues/10757

**Comment by urbalazs** [comment]
Not a PHP warning, but a database error.

```
2022-02-07T23:07:44Z index [ERROR]: DB Error {
  "code": 1062,
  "error": "Duplicate entry 'http://blimey.social' for key 'nurl'",
  "callstack": "DBA::insert, GServer::detect, GServer::check, GServer::getID, APContact::getByURL, Receiver::getReceivers, Receiver::processObject, Receiver::fetchObject",
  "params": "INSERT INTO `gserver` (`detection-method`, `register_policy`, `platform`, `version`, `site_name`, `registered-users`, `network`, `directory-type`, `poco`, `url`, `nurl`, `next_contact`, `last_contact`, `failed`, `created`) VALUES (102, 2, 'pleroma', '2.4.2', 'Blimey Social', 4, 'apub', 0, '', 'https://blimey.social', 'http://blimey.social', '2022-02-14 23:07:43', '2022-02-07 23:07:43', 0, '2022-02-07 23:07:43')"
} - {
  "file": "Database.php",
  "line": 771,
  "function": "e",
  "uid": "c5bf73",
  "process_id": 108761
}
```

PHP 7.4.3
MariaDB 10.3.32
Friendica 2022.02

**Comment by urbalazs** [comment]
Another database error.

```
2022-02-08T21:43:33Z index [ERROR]: DB Error {
  "code": 1406,
  "error": "Data too long for column 'preview' at row 1",
  "callstack": "Item::insert, Processor::postItem, Receiver::processActivity, Processor::fetchMissingActivity, Processor::createItem, Receiver::processActivity, Receiver::processInbox, Inbox::rawContent",
  "params": "INSERT INTO `post-media` (`uri-id`, `type`, `url`, `mimetype`, `size`, `preview`, `preview-height`, `preview-width`, `description`, `name`, `author-url`, `author-name`, `author-image`, `publisher-url`, `publisher-name`, `publisher-image`) VALUES (49265, 17, 'https://www.theguardian.com/world/2022/feb/04/rotterdam-says-no-request-made-to-dismantle-bridge-for-bezoss-yacht-reports', 'text/html; charset=UTF-8', 237473, 'https://i.guim.co.uk/img/media/d0bcd55daf9ef8dc1491091360bb42e4cb35116a/0_300_9000_5400/master/9000.jpg?width=1200&height=630&quality=85&auto=format&fit=crop&overlay-align=bottom%2Cleft&overlay-width=100p&overlay-base64=L2ltZy9zdGF0aWMvb3ZlcmxheXMvdGctZGVmYXVsdC5wbmc&enable=upscale&s=abd9c61877d990f15d0ce83c809852e2', 630, 1200, 'Mayor denies any decision to accommodate the billionaire Amazon boss, despite claims that historic bridge will be removed', 'Rotterdam says ‘no request’ to dismantle bridge for Bezos’s yacht – reports', NULL, 'Guardian staff reporter', NULL, 'https://www.theguardian.com/', 'The Guardian', 'https://uploads.guim.co.uk/2018/01/31/TheGuardian_AMP.png') ON DUPLICATE KEY UPDATE `uri-id` = 49265, `type` = 17, `url` = 'https://www.theguardian.com/world/2022/feb/04/rotterdam-says-no-request-made-to-dismantle-bridge-for-bezoss-yacht-reports', `mimetype` = 'text/html; charset=UTF-8', `size` = 237473, `preview` = 'https://i.guim.co.uk/img/media/d0bcd55daf9ef8dc1491091360bb42e4cb35116a/0_300_9000_5400/master/9000.jpg?width=1200&height=630&quality=85&auto=format&fit=crop&overlay-align=bottom%2Cleft&overlay-width=100p&overlay-base64=L2ltZy9zdGF0aWMvb3ZlcmxheXMvdGctZGVmYXVsdC5wbmc&enable=upscale&s=abd9c61877d990f15d0ce83c809852e2', `preview-height` = 630, `preview-width` = 1200, `description` = 'Mayor denies any decision to accommodate the billionaire Amazon boss, despite claims that historic bridge will be removed', `name` = 'Rotterdam says ‘no request’ to dismantle bridge for Bezos’s yacht – reports', `author-url` = NULL, `author-name` = 'Guardian staff reporter', `author-image` = NULL, `publisher-url` = 'https://www.theguardian.com/', `publisher-name` = 'The Guardian', `publisher-image` = 'https://uploads.guim.co.uk/2018/01/31/TheGuardian_AMP.png'"
} - {
  "file":"Database.php",
  "line":771,
  "function":"e",
  "uid":"841c2f",
  "process_id":109392
}
```

```
2022-02-08T21:43:29Z index [ERROR]: DB Error {
  "code": 1406,
  "error": "Data too long for column 'k' at row 1",
  "callstack": "Database::update, DatabaseCache::set, Images::getInfoFromURLCached, ParseUrl::Friendica\\Util\\{closure}, array_walk, Item::insert, Processor::postItem, Receiver::processActivity",
  "params": "REPLACE `cache` (`k`, `v`, `expires`, `updated`) VALUES ('https://i.guim.co.uk/img/media/d0bcd55daf9ef8dc1491091360bb42e4cb35116a/0_300_9000_5400/master/9000.jpg?width=1200&height=630&quality=85&auto=format&fit=crop&overlay-align=bottom%2Cleft&overlay-width=100p&overlay-base64=L2ltZy9zdGF0aWMvb3ZlcmxheXMvdGctZGVmYXVsdC5wbmc&enable=upscale&s=abd9c61877d990f15d0ce83c809852e2', 'a:8:{i:0;i:1200;i:1;i:630;i:2;i:2;i:3;s:25:\\\"width=\\\"1200\\\" height=\\\"630\\\"\\\";s:4:\\\"bits\\\";i:8;s:8:\\\"channels\\\";i:3;s:4:\\\"mime\\\";s:10:\\\"image/jpeg\\\";s:4:\\\"size\\\";i:87078;}', '2022-02-08 21:48:29', '2022-02-08 21:43:29')"
} - {
  "file":"Database.php",
  "line":771,
  "function":"e",
  "uid":"841c2f",
  "process_id":109392
}
```

```
2022-02-11T04:55:21Z index [ERROR]: DB Error {
  "code": 1406,
  "error": "Data too long for column 'name' at row 1",
  "callstack": "Item::insert, Processor::postItem, Receiver::processActivity, Processor::fetchMissingActivity, Processor::createItem, Receiver::processActivity, Processor::fetchMissingActivity, Processor::createItem",
  "params": "INSERT INTO `post-media` (`uri-id`, `type`, `url`, `mimetype`, `description`, `name`, `author-url`, `author-name`, `author-image`, `publisher-url`, `publisher-name`, `publisher-image`) VALUES (49697, 17, 'https://www.reddit.com/r/NoMansSkyTheGame/comments/jad3wc/snow_spirits_this_8_page_journal_contains_details/', 'text/html; charset=utf-8', '243 votes and 34 comments so far on Reddit', 'r/NoMansSkyTheGame - SNOW SPIRITS - This 8 page journal contains details of my study of the Snow Spirits. I am a traveler scientist who studies this type of creature. (This took nearly 3 hours to complete, but I think it was worth the effort). I hope you all enjoy learning about the mysterious Snow Spirits!', NULL, NULL, NULL, NULL, 'reddit', NULL) ON DUPLICATE KEY UPDATE `uri-id` = 49697, `type` = 17, `url` = 'https://www.reddit.com/r/NoMansSkyTheGame/comments/jad3wc/snow_spirits_this_8_page_journal_contains_details/', `mimetype` = 'text/html; charset=utf-8', `description` = '243 votes and 34 comments so far on Reddit', `name` = 'r/NoMansSkyTheGame - SNOW SPIRITS - This 8 page journal contains details of my study of the Snow Spirits. I am a traveler scientist who studies this type of creature. (This took nearly 3 hours to complete, but I think it was worth the effort). I hope you all enjoy learning about the mysterious Snow Spirits!', `author-url` = NULL, `author-name` = NULL, `author-image` = NULL, `publisher-url` = NULL, `publisher-name` = 'reddit', `publisher-image` = NULL"
} - {
  "file":"Database.php",
  "line":771,
  "function":"e",
  "uid":"15e269",
  "process_id":137508
}
```

**Comment by urbalazs** [comment]
Not a PHP but a worker warning.

```
2022-02-11T08:20:06Z worker [WARNING]: Unexpected data {
  "outbox": {
    "@context": [
      "https://www.w3.org/ns/activitystreams",
      "https://w3id.org/security/v1",
      {
        "addressRegion": "sc:addressRegion",
        "timezone": {
          "@id": "mz:timezone",
          "@type": "sc:Text"
        },
        "pt": "https://joinpeertube.org/ns#",
        "manuallyApprovesFollowers": "as:manuallyApprovesFollowers",
        "inLanguage": "sc:inLanguage",
        "address": {
          "@id": "sc:address",
          "@type": "sc:PostalAddress"
        },
        "discoverable": "toot:discoverable",
        "repliesModerationOption": {
          "@id": "mz:repliesModerationOption",
          "@type": "mz:repliesModerationOptionType"
        },
        "sc": "http://schema.org#",
        "mz": "https://joinmobilizon.org/ns#",
        "category": "sc:category",
        "joinModeType": {
          "@id": "mz:joinModeType",
          "@type": "rdfs:Class"
        },
        "Hashtag": "as:Hashtag",
        "propertyID": "sc:propertyID",
        "PostalAddress": "sc:PostalAddress",
        "discussions": {
          "@id": "mz:discussions",
          "@type": "@id"
        },
        "streetAddress": "sc:streetAddress",
        "anonymousParticipationEnabled": {
          "@id": "mz:anonymousParticipationEnabled",
          "@type": "sc:Boolean"
        },
        "addressLocality": "sc:addressLocality",
        "joinMode": {
          "@id": "mz:joinMode",
          "@type": "mz:joinModeType"
        },
        "location": {
          "@id": "sc:location",
          "@type": "sc:Place"
        },
        "toot": "http://joinmastodon.org/ns#",
        "uuid": "sc:identifier",
        "maximumAttendeeCapacity": "sc:maximumAttendeeCapacity",
        "participationMessage": {
          "@id": "mz:participationMessage",
          "@type": "sc:Text"
        },
        "openness": {
          "@id": "mz:openness",
          "@type": "@id"
        },
        "members": {
          "@id": "mz:members",
          "@type": "@id"
        },
        "events": {
          "@id": "mz:events",
          "@type": "@id"
        },
        "resources": {
          "@id": "mz:resources",
          "@type": "@id"
        },
        "addressCountry": "sc:addressCountry",
        "posts": {
          "@id": "mz:posts",
          "@type": "@id"
        },
        "commentsEnabled": {
          "@id": "pt:commentsEnabled",
          "@type": "sc:Boolean"
        },
        "value": "sc:value",
        "PropertyValue": "sc:PropertyValue",
        "repliesModerationOptionType": {
          "@id": "mz:repliesModerationOptionType",
          "@type": "rdfs:Class"
        },
        "todos": {
          "@id": "mz:todos",
          "@type": "@id"
        },
        "ical": "http://www.w3.org/2002/12/cal/ical#",
        "postalCode": "sc:postalCode",
        "@language": "und"
      }
    ],
    "attributedTo": "https://mobilizon.picasoft.net/@bkil",
    "first": {
      "attributedTo": "https://mobilizon.picasoft.net/@bkil",
      "id": "https://mobilizon.picasoft.net/@bkil/outbox?page=1",
      "orderedItems": [],
      "partOf": "https://mobilizon.picasoft.net/@bkil/outbox",
      "type": "OrderedCollectionPage"
    },
    "id": "https://mobilizon.picasoft.net/@bkil/outbox",
    "totalItems": 0,
    "type": "OrderedCollection"
  },
  "worker_id": "7e50d6b",
  "worker_cmd": "UpdateContact"
} - {
  "file": "Probe.php",
  "line": 2087,
  "function": "updateFromOutbox",
  "uid": "b49fc9",
  "process_id": 163321
}
```

**Comment by annando** [comment]
@urbalazs this issue is especially for PHP warnings. It would be great if you could report other messages in different issues. Otherwise this issue here would be rapidly flooded with different messages of all kind, thanks!

**Comment by tobiasd** [comment]
```
PHP Warning:  strtotime(): Epoch doesn't fit in a PHP integer in /src/Model/Contact.php on line 1785
```

**Comment by MrPetovan** [comment]
Here it is, the fabled Year 2K22 Bug! Also I'm surprised someone is running PHP on a 16-bit architecture, but I'm not surprised it's you of all the people. Raspberry Pi, I presume?

**Comment by tobiasd** [comment]
Yes, it's a RasPi and I've not yet reinstalled the system with the new 64bit Raspbian

**Comment by MrPetovan** [comment]
Got it, unfortunately I'm going to have to draw the line at supporting a minimum of 32-bit architectures since we are heavily relying on the native PHP `strtotime()` function and I'm not even sure what it should be replaced with and you'd be the only one able to test the fix.

**Comment by tobiasd** [comment]
If that is the reason for the  warning, I'm fine with neglecting it and not providing a 32bit version of Friendica :)

**Comment by MrPetovan** [comment]
Yes, this is the reason for the warning.

**Comment by ghost** [comment]
latest dev
`PHP Warning:  DOMXPath::query(): Undefined namespace prefix in /friendica/src/Util/XML.php on line 478`

**Comment by tobiasd** [comment]
> If that is the reason for the warning, I'm fine with neglecting it and not providing a 32bit version of Friendica :)

I've switched to the 64bit kernel, so lets see if the warning persists.

**Comment by ghost** [comment]
> PHP Warning: DOMXPath::query(): Undefined namespace prefix in /friendica/src/Util/XML.php on line 478

Is there already a solution for this message?

**Comment by MrPetovan** [comment]
We would need to know about the offending XML document to fix this issue. I see it too on my production node.

**Comment by ghost** [comment]
Latest Dev

`PHP Warning:  sem_acquire(): failed to acquire key 0x6603003e: Identifier removed in /src/Core/Lock/Type/SemaphoreLock.php on line 65`

**Comment by AlfredSK** [comment]
2022.05-dev

```
PHP Notice:  Undefined variable: old_featured in .../src/Protocol/ActivityPub/Processor.php on line 1011
PHP Warning:  Invalid argument supplied for foreach() in .../src/Protocol/ActivityPub/Processor.php on line 1011
```

**Comment by AlfredSK** [comment]
2022.05-dev

```
PHP Warning:  array_search() expects parameter 2 to be array, null given in .../src/Protocol/ActivityPub/Processor.php on line 1038
```

**Comment by tobiasd** [comment]
2022.05-rc b5129eb4ed0ad2c4a3f3d983d0ea70e4b63506b0

```
PHP Warning:  Undefined array key "network" in /src/Module/Search/Acl.php on line 83
PHP Warning:  Undefined array key "url" in /src/Module/Search/Acl.php on line 84
PHP Warning:  Undefined array key "contact-type" in /src/Module/Search/Acl.php on line 85
```

**Comment by tobiasd** [comment]
2022.05-rc b5129eb4ed0ad2c4a3f3d983d0ea70e4b63506b0

```
PHP Warning:  Undefined array key "url" in /src/Model/Contact.php on line 1703
```
Addressed in #11494
